apps/dsb-client-gateway-api/src/app/modules/message/service/message.service.ts

```diff
@@ -471,26 +475,58 @@ export class MessageService {
       '0x' + privateKey
     );
 
-    await this.sendSymmetricKeys(
-      this.logger,
-      qualifiedDids,
-      symmetricKey,
-      clientGatewayMessageId
-    );
-
-    //uploading file
-    const result: SendMessageResponseFile =
-      await this.ddhubFilesService.uploadFile(
-        fs.createReadStream(filePath),
-        file.originalname,
+    if (channel.payloadEncryption) {
+      await this.sendSymmetricKeys(
+        this.logger,
         qualifiedDids,
-        topic.id,
-        topic.version,
-        signature,
-        clientGatewayMessageId,
-        channel.payloadEncryption,
-        dto.transactionId
+        symmetricKey,
+        clientGatewayMessageId
       );
+    }
+
+    let result: SendMessageResponseFile = null;
+    if (this.enableUploadChunk) {
+      const _file = await buffer(fs.createReadStream(filePath));
```

Comment:
it will cause high memory consumption as you read whole stream into a buffer - could cause buffer overflow

Reply:
yup, any suggestion. how encryption handle it ?

Reply:
streams

https://github.com/dannycho7/split-file-stream

you can use this library - split file to smaller chunks, upload them one by one. Calculate `maxFileSize` dynamically using some env. variable